src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py

```diff
@@ -129,3 +130,99 @@ def auto_register(func, *args, **kwargs):
             raise CLIInternalError('Registeration failed, please manually run command '
                                    '`az provider register -n Microsoft.ServiceLinker` to register the provider.')
         raise ex
+
+
+def create_key_vault_reference_connection_if_not_exist(cmd, client, source_id, key_vault_id):
+    from ._validators import get_source_resource_name
+    from knack.log import get_logger
+    logger = get_logger(__name__)
+
+    logger.warning('get valid key vualt reference connection')
+    all_connections = run_cli_cmd('az webapp connection list --source-id {} -o json'.format(source_id))
+    key_vault_connections = []
+    for connection in all_connections:  # pylint: disable=not-an-iterable
+        if connection.get('targetId') == key_vault_id:
+            key_vault_connections.append(connection)
+
+    source_name = get_source_resource_name(cmd)
+    auth_info = get_auth_if_no_valid_key_vault_connection(logger, source_name, source_id, key_vault_connections)
+    if not auth_info:
+        return
+
+    # No Valid Key Vault Connection, Create
+    logger.warning('no valid key vault connection found. Creating...')
+    client = set_user_token_header(client, cmd.cli_ctx)
+
+    from ._resource_config import CLIENT_TYPE
+
+    connection_name = generate_random_string(prefix='keyvault_')
+    parameters = {
+        'target_id': key_vault_id,
+        'auth_info': auth_info,
+        'client_type': CLIENT_TYPE.Dotnet,  # Key Vault Configuration are same across all client types
+    }
+
+    return auto_register(client.begin_create_or_update,
+                         resource_uri=source_id,
+                         linker_name=connection_name,
+                         parameters=parameters)
+
+
+def get_auth_if_no_valid_key_vault_connection(logger, source_name, source_id, key_vault_connections):
+    auth_type = 'systemAssignedIdentity'
+    client_id = None
+    subscription_id = None
+
+    if len(key_vault_connections) > 0:
+        from ._resource_config import RESOURCE
+        from msrestazure.tools import (
+            parse_resource_id,
+            is_valid_resource_id
+        )
+
+        # https://docs.microsoft.com/azure/app-service/app-service-key-vault-references
+        if source_name == RESOURCE.WebApp:
+            try:
+                webapp = run_cli_cmd('az rest -u {}?api-version=2020-09-01 -o json'.format(source_id))
```

Comment:
why not use rest call via mgmt sdk directly

Reply:
Backgound is SC has too many Azure service dependencies, so we prefer to call CLI command directly to avoid too many SDK dependencies, making it simple.